intel-retail/automated-self-checkout
# feat: refactor grpc_python profile and make it run inside docker

## PR Checklist



## What are you changing?


## Issue this PR will close

close: #**issue_number**

## Anything the reviewer should know when reviewing this PR?

## Test Instructions if applicable

- git clone this PR
- make build-grpc-python
- for **run.sh** test:
```console
$ ./camera-simulator/camera-simualtor.sh
$ PIPELINE_PROFILE="grpc_python" sudo -E ./run.sh --workload ovms --platform core  inputsrc rtsp://127.0.0.1:8554/camera_0
```
you should see the .profile-launcher kicks off the grpc_python profile and runs it.
you can use `make clean-grpc-python` to clean up the running stuff since this profile will run forever.
then you can exam the ./result/r0.jsonl file to see the outputs

- for **stream-density** test:
```console
$ cd ./benchmark-scripts/
$ PIPELINE_PROFILE="grpc_python" sudo -E ./benchmark.sh --stream_density 15.0 --workload ovms --platform core --inputsrc rtsp://127.0.0.1:8554/camera_0 --logdir grpc_python/data --duration 60 --init_duration 20
```
after the stream_density is finished, you can exam the stream_density.log file under ./results folder.



## If the there are associated PRs in other repositories, please link them here (i.e. intel-retail/automated-self-checkout )
